### PR TITLE
feat(workflow): Adding project slugs to Rule and AlertRule serializer

### PR DESCRIPTION
**Pull request by iProgramStuff**
Adding project slugs to the api response so the frontend can display them in the rule listing.

### CODE REVIEW

**Review by scttcper — APPROVED** [review]
lgtm 👍

File: src/sentry/api/serializers/models/alert_rule.py [hunk 1]
```diff
@@ -28,6 +28,20 @@ def get_attrs(self, item_list, user, **kwargs):
...
+        alert_rules = {item.id: item for item in item_list}
+        for alert_rule_id, project_slug in alert_rule_projects:
+            rule_result = result[alert_rules[alert_rule_id]].setdefault("projects", [])
+            rule_result.append(project_slug)
+
+        for alert_rule_id, project_slug in AlertRuleExcludedProjects.objects.filter(
+            alert_rule__in=item_list
+        ).values_list("alert_rule_id", "project__slug"):
+            exclusions = result[alert_rules[alert_rule_id]].setdefault("excludedProjects", [])
+            exclusions.append(project_slug)
```
Comment (wedamija) [inline src/sentry/api/serializers/models/alert_rule.py:43]: Do we need the excluded projects here? I'd leave them in the detailed serializer otherwise.

Comment (iProgramStuff) [reply src/sentry/api/serializers/models/alert_rule.py:43]: Hmm, doesn't it seem appropriate? We could use the info to know which projects to display in the list (probably more useful once we have multi project alerts?). 

Could put it back into the detailed serializer, but bringing it in here also makes the details serializer redundant (I was able to remove it)

Comment (wedamija) [reply src/sentry/api/serializers/models/alert_rule.py:43]: We have a general problem where we start putting more and more things into a serializer, and at some point our endpoints end up slow due to extra serialization + db queries. Unless we actually need this data now I'd say it should remain in the detailed serializer.

Comment (iProgramStuff) [reply src/sentry/api/serializers/models/alert_rule.py:43]: Gotcha. Will put it back into the detailed serializer until we have an explicit need.

**Pull request closed without merging**